extensions/amp-date-picker/1.0/component.js

```diff
@@ -0,0 +1,109 @@
+import * as Preact from '#preact';
+import {useMemo} from '#preact';
+
+import './amp-date-picker.css';
+import {
+  DEFAULT_END_INPUT_SELECTOR,
+  DEFAULT_INPUT_SELECTOR,
+  DEFAULT_MONTH_FORMAT,
+  DEFAULT_ON_ERROR,
+  DEFAULT_START_INPUT_SELECTOR,
+  DEFAULT_WEEK_DAY_FORMAT,
+  DatePickerMode,
+  DatePickerType,
+  ISO_8601,
+} from './constants';
+import {getCurrentDate} from './date-helpers';
+import {DateRangePicker} from './date-range-picker';
+import {DatesList} from './dates-list';
+import {SingleDatePicker} from './single-date-picker';
+import {AttributesContext} from './use-attributes';
+
+/**
+ * @param {!BentoDatePicker.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function BentoDatePicker({
+  children,
+  type = DatePickerType.SINGLE,
+  mode = DatePickerMode.STATIC,
+  inputSelector = DEFAULT_INPUT_SELECTOR,
+  startInputSelector = DEFAULT_START_INPUT_SELECTOR,
+  endInputSelector = DEFAULT_END_INPUT_SELECTOR,
+  format = ISO_8601,
+  id,
+  onError = DEFAULT_ON_ERROR,
+  initialVisibleMonth,
+  blocked,
+  allowBlockedRanges,
+  allowBlockedEndDate,
+  highlighted,
+  min = getCurrentDate(),
+  max,
+  monthFormat = DEFAULT_MONTH_FORMAT,
+  weekDayFormat = DEFAULT_WEEK_DAY_FORMAT,
+}) {
+  const blockedDates = useMemo(() => {
+    return new DatesList(blocked);
+  }, [blocked]);
+
+  const highlightedDates = useMemo(() => {
+    return new DatesList(highlighted);
+  }, [highlighted]);
+
+  const datePicker = useMemo(() => {
+    const commonProps = {
+      blockedDates,
+      children,
+      format,
+      monthFormat,
+      id,
+      initialVisibleMonth,
+      mode,
+      onError,
+      weekDayFormat,
+    };
+    if (type === DatePickerType.SINGLE) {
+      const props = {
+        ...commonProps,
+        inputSelector,
+      };
+      return <SingleDatePicker {...props} />;
+    } else if (type === DatePickerType.RANGE) {
+      const props = {
+        ...commonProps,
+        startInputSelector,
+        endInputSelector,
+        allowBlockedEndDate,
+        allowBlockedRanges,
+      };
+      return <DateRangePicker {...props} />;
+    } else {
+      onError('Invalid date picker type');
+    }
+  }, [
+    blockedDates,
+    format,
+    id,
+    initialVisibleMonth,
+    mode,
+    onError,
+    type,
+    allowBlockedEndDate,
+    allowBlockedRanges,
+    startInputSelector,
+    endInputSelector,
+    inputSelector,
+    children,
+    monthFormat,
+    weekDayFormat,
+  ]);
+
+  return (
+    <AttributesContext.Provider
+      value={{blockedDates, highlightedDates, allowBlockedEndDate, min, max}}
+    >
+      {datePicker}
+    </AttributesContext.Provider>
+  );
```

Comment:
Just a suggestion. It does not need to be done like this. But what if we make datePicker an alias to one of the DatePicker components, then just passed it all the props. It should just ignore any props that it does not need.
```suggestion
  const DatePicker = useMemo(() => {
    switch(type) {
      case DatePickerType.SINGLE:
        return SingleDatePicker;
      case DatePickerType.RANGE:
        return DateRangePicker;
      default:
        return onError('Invalid date picker type');
    }
  }, [type]);

  return (
    <AttributesContext.Provider
      value={{blockedDates, highlightedDates, allowBlockedEndDate, min, max}}
    >
      <DatePicker {...props}/>
    </AttributesContext.Provider>
  );
```
or something like that. I think it might be a bit easier to read. But that's just me.